### ISSUE TITLE
Prefer using Events to notify end of animations

### ISSUE THREAD

**Issue by Kerollmops** [jcornaz/benimator#36]
I found this incredible crate and use it for some cool game dev, it was really fun!

However, I also worked with the newly released `bevy_tweening` crate and [needed a way to detect the end of animations to do some stuff](https://github.com/djeedai/bevy_tweening/issues/3). I proposed that we could use the same technique as you are using: removing a `Play` component. Unfortunately, we asked some ECS peeps about doing that compared to using `Events` instead, [their answer was that we should prefer using `Events` for performances reasons](https://discord.com/channels/691052431525675048/742569353878437978/943262159658418196).

I know that the `Play` component plays more than this role of detecting the end of an animation, but maybe you would be interested in this. As I understand it is preferable to avoid adding/removing components too much from the ECS and prefer doing things like in `bevy_tweening` by using the `Animator` and changing its state.

**Comment by jcornaz** [comment]
Hi, thanks for the suggestion :-)

Yes, using events is better for performance. Though the question is not really "should we use event *instead of* a `Play` component", because we could use both (like keeping the `Play` component being the source of truth for playing the animation, and sending an event to notify when the animation is ended). The question is more "Is it good to use a `Play` component to define if an animation is running".

But before going into performances, my main concern is ergonomy. And I think relying on the presence of a component makes it easy to query alongside other components. Like I can easily get all enemies that are animated, or all vehicles that are not animated, and directly access the other components too.

Then when it comes to performance, I think there are two important things to consider:
1. We need to first demonstrate an actual, *noticeable* performance problem that is worth solving. Just because it is "in theory" slower, or because it takes 100ns instead of 10ns is not enough to deteriorate the ergonomy in my opinion. Because this is rust and bevy, there is a good chance it is already much faster than in other game-engines and languages anyway.
2. We have to consider the overall performance, not only the insertion/removal of the component. Yes, it is slow to insert and remove the `Play` component. But that is not happening in most frames. What is happening in each frame is the fetch all animations that are in progress. Without the `Play` component we may have to iterate over more entities, and we should make sure that cost is worth the benefits.

Lastly, still in the mod of cost-benefit evaluation, we have to consider that it may require a breaking change. And if it does, then it must come with pretty high benefits.


So, to summarize: I am open to it. But, I first want to observe a clearly compelling cost-benefit balance.


*P.S: Make sure to use the version [2.0.1](https://github.com/jcornaz/benimator/releases/tag/v2.0.1) which actually contains an optimization to make the play component faster to insert/remove.*
